src/vs/code/electron-browser/sharedProcess/sharedProcess.js

```diff
@@ -6,6 +6,10 @@
 //@ts-check
 'use strict';
 
+// Set the below before requiring applicationinsights in the shared process
+process.env['APPLICATION_INSIGHTS_NO_DIAGNOSTIC_CHANNEL'] = 'true'; // Skip monkey patching of 3rd party modules by appinsights
+global['diagnosticsSource'] = {}; // Prevents diagnostic channel (which patches "require") from initializing entirely
+
```

Comment:
@bpasero This is the workaround to avoid the monkey patching that gets done by the newer version of application insights. It needs to be placed before application insights module gets imported. Based on the way we load things, this is definitely the safest place for the workaround, but maybe not the best. 

Do you have any other ideas for the right place for this?

Reply:
@ramya-rao-a you can add a helper method to bootstrap.js and then call it from where we need it, looks like:
* `src/cli.js`
* `src/vs/code/electron-browser/sharedProcess/sharedProcess.js`

Reply:
@bpasero Pushed a commit to make the change as you suggested.

It works as expected for the shared process, but not for the cli. 
I added some console logs in the modules that do the patching and below is what I see when running the cli

![image](https://user-images.githubusercontent.com/16890566/45896736-dd8d6300-bd89-11e8-8a07-8afc924e0a9f.png)

The application insights module gets loaded even before https://github.com/Microsoft/vscode/blob/master/scripts/code-cli.bat#L24

Thoughts?